action.yml

```diff
@@ -0,0 +1,82 @@
+name: Configure AWS Profile
+
+description: Configures an OIDC AWS Role Session as an aws config profile.
+
+
+inputs:
+  role-to-assume:
+    description: "ARN of the role to assume."
+    required: true
+  aws-profile-name:
+    description: "Name of profile to be created. If not set ID of step is used."
+    required: false
+  role-duration-seconds:
+    description: "Role duration in seconds. For role chaining max: 3600."
+    default: "7200"
+    required: false
+  aws-region:
+    description: "Name of AWS region to be used."
+    default: "eu-west-1"
+    required: false
+  aws-access-key-id:
+    description: "AWS Access Key ID."
+    required: false
+  aws-secret-access-key:
+    description: "AWS Secret Access Key."
+    required: false
+  aws-session-token:
+    description: "AWS Session Token"
+    required: false
+  aws-source-profile:
+    description: "Name of source profile for assuming role by SDKs"
+    required: false
+
+outputs:
+  aws-profile:
+    description: "Created AWS profile name"
+    value: ${{ inputs.aws-profile-name || github.action }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Profile Based on Source Profile
+      shell: bash
+      if: inputs.aws-source-profile
+      run: |
+        PROFILE_NAME=${{ inputs.aws-profile-name || github.action }}
```

Comment:
maybe move **PROFILE_NAME** to action env and then use same value instead of local variable - also in output

Reply:
can be improved later, I need to have some version merged where to start with